tests/exporters/test_onnx_export.py

```diff
@@ -266,6 +270,69 @@ def _onnx_export(self, test_name, name, model_name, task, onnx_config_class_cons
             except (RuntimeError, ValueError) as e:
                 self.fail(f"{name}, {task} -> {e}")
 
+    def _onnx_export_for_encoder_decoder_models_for_ort(
+        self, test_name, name, model_name, task, onnx_config_class_constructor, device="cpu"
+    ):
+        model_class = TasksManager.get_model_class_for_task(task)
+        config = AutoConfig.from_pretrained(model_name)
+        model = model_class.from_config(config)
+
+        onnx_config = onnx_config_class_constructor(model.config)
+
+        if is_torch_available():
+            from optimum.exporters.onnx.utils import TORCH_VERSION
+
+            if not onnx_config.is_torch_support_available:
+                pytest.skip(
+                    "Skipping due to incompatible PyTorch version. Minimum required is"
+                    f" {onnx_config.MIN_TORCH_VERSION}, got: {TORCH_VERSION}"
+                )
+
+        encoder_model = model.get_encoder()
+        encoder_onnx_config = onnx_config.get_encoder_onnx_config(encoder_model.config)
+
+        decoder_model = model.get_decoder()
+        decoder_onnx_config = onnx_config.get_decoder_onnx_config(
+            decoder_model.config, task.replace("-with-past", ""), use_past=False
+        )
```

Comment:
I think this is not neat, we need to either make sure that when `task` does not contain `"with-past"` we have `use_past=False` (isn't it the case already?), or that setting `use_past` to `False`, would ignore the `"with-past"` in `task`, wdyt?

Reply:
Thought so. This function is used to return the  `decoder_onnx_config` object. `OnnxConfig` initalizer do not accept `task` with `-with-past` suffix or it would throw an error. Same is being applied here: [tasks.py:101](https://github.com/huggingface/optimum/blob/main/optimum/exporters/tasks.py#L101).

```python
return SpeechSeq2SeqDecoderOnnxConfig(config, task, use_past=use_past)
```

But in this specific case, we would want to export models with `use_past=True` and `use_past=False` when `-with-past` is present in the task.  Hence, we anyways would want to remove the `with-past` from task explicitly.